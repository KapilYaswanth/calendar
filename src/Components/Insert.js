import React,{useState} from 'react'
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
const useStyles = makeStyles((theme) => ({
    root: {
      '& > *': {
        margin: theme.spacing(1),
        width: '25ch',
      },
    },
  }));
export default function Insert() {
    const classes = useStyles();
    const [name, setname] = useState("")
    const [start, setstart] = useState("")
    const [end, setend] = useState("")
    const handleClick = () => {
    var gapi = window.gapi
    var CLIENT_ID = "639100172651-75tnqt7k8b64d75n8en3ghosh8jnh9d9.apps.googleusercontent.com"
    var API_KEY = "AIzaSyBChZ90Wv3zs6WMlkgW75EZbobmbsO8ePs"
    var SCOPES = "https://www.googleapis.com/auth/calendar.events"
        gapi.load('client:auth2', () => {
          console.log('loaded client')
    
          gapi.client.init({
            apiKey: API_KEY,
            clientId: CLIENT_ID,
            scope: SCOPES,
          })
    
    
          gapi.auth2.getAuthInstance().signIn()
          .then(() => {
            var event = {
              'summary': name,
              'start': {
                'date': start,
                'timeZone': 'Asia/Kolkata'
              },
              'end': {
                'date': end,
                'timeZone': 'Asia/Kolkata'
              },
              'hangoutLink': 'https://meet.google.com/rph-dxck-cxu',
           }
            var request = gapi.client.calendar.events.insert({
              'calendarId': 'primary',
              'resource': event,
            })
    
            request.execute(event => {
              console.log(event)
              
            })
          })
        })
      }
    
    return (
        
        <div>
            <form data-testid="insert form" onSubmit={(event)=>{
                event.preventDefault();
                handleClick();
                setname("");
                setstart("");
                setend("");
            }} className={classes.root}>
        <TextField data-testid="Input event name" value={name} id="outlined-basic" placeholder="Event Name" label="Event Name" variant="outlined" onChange={(e)=>setname(e.target.value)} required/><br/>
        <TextField id="outlined-basic" value={start} label="Start Date  yyyy-MM-dd" placeholder="Input Start Date" variant="outlined" onChange={(e)=>setstart(e.target.value)}required/><br/>
        <TextField id="outlined-basic" value={end} label="End Date  yyyy-MM-dd" placeholder="Input End Date" variant="outlined" onChange={(e)=>setend(e.target.value)}required/>
        <div><button type="submit" className="button" >Add Event</button></div>
      </form>
      
        </div>
    )
}
