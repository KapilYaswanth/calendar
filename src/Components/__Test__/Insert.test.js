import React from 'react';
import {render,fireEvent, cleanup} from '@testing-library/react';
import Insert from '../Insert';
import renderer from 'react-test-renderer'

afterEach(cleanup);

it("renders correctly",()=>{
    const {queryByTestId} = render(<Insert />)
    expect(queryByTestId("insert form")).toBeTruthy();
})

describe("Input Event name",() => {
    it("updates on change",() => {
        const {queryByPlaceholderText} = render(<Insert />)
        const textField = queryByPlaceholderText("Event Name");
        fireEvent.change(textField,{target:{value:"test"}})
        expect(textField.value).toBe("test")
    })
})

describe("Input Start Date",() => {
    it("updates on change",() => {
        const {queryByPlaceholderText} = render(<Insert />)
        const textField = queryByPlaceholderText("Input Start Date");
        fireEvent.change(textField,{target:{value:"test"}})
        expect(textField.value).toBe("test")
    })
})

it("Create Insert Form Matches Snapshot",()=>{
    const tree = renderer.create(<Insert/>).toJSON();
    expect(tree).toMatchSnapshot();
})

