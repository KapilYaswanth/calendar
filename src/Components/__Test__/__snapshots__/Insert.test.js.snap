// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Insert Form Matches Snapshot 1`] = `
<div>
  <form
    className="makeStyles-root-16"
    data-testid="insert form"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
      data-testid="Input event name"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="outlined-basic"
        id="outlined-basic-label"
      >
        Event Name
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="outlined-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Event Name"
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-19"
          >
            <span>
              Event Name
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="outlined-basic"
        id="outlined-basic-label"
      >
        Start Date  yyyy-MM-dd
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="outlined-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Input Start Date"
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-19"
          >
            <span>
              Start Date  yyyy-MM-dd
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="outlined-basic"
        id="outlined-basic-label"
      >
        End Date  yyyy-MM-dd
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="outlined-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Input End Date"
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-19"
          >
            <span>
              End Date  yyyy-MM-dd
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div>
      <button
        className="button"
        type="submit"
      >
        Add Event
      </button>
    </div>
  </form>
</div>
`;
